ui-react/src/components/topBar/ReportProblemModal.tsx

```diff
@@ -0,0 +1,248 @@
+import React, { useState } from "react";
+import { Button, Form, Modal, Row, Col, Stack } from "react-bootstrap";
+import { openExternal } from "../../eventsBridge";
+import { repository } from "../../../package.json";
+import os from "os";
+import LogsCanvas from "./LogsCanvas";
+
+type ReportProblemForm = {
+  title?: string;
+  email?: string;
+  details?: string;
+};
+
+type ReportProblemModalProps = {
+  show: boolean;
+  handleCloseModal: () => {};
+};
+
+function ReportProblemModal({
+  show,
+  handleCloseModal,
+}: ReportProblemModalProps) {
+  const [form, setForm] = useState<ReportProblemForm>({
+    title: "",
+    email: "",
+    details: "",
+  });
+
+  const [errors, setErrors] = useState({});
+  const [showLogs, setShowLogs] = useState(false);
+
+  const setField = (field: string, value: string) => {
+    setForm({ ...form, [field]: value });
+
+    if (!!errors[field]) setErrors({ ...errors, [field]: null });
+  };
+
+  const validateForm = (validateEmail = true) => {
+    const newErrors = {};
+
+    if (!form.title || form.title === "") newErrors.title = "שדה חובה";
+
+    if (validateEmail) {
+      if (!form.email || !isValidEmail(form.email))
+        newErrors.email = "שדה מייל לא חוקי";
+    }
+
+    return newErrors;
+  };
+
+  function isValidEmail(email) {
+    return /\S+@\S+\.\S+/.test(email);
+  }
```

Comment:
This function need to be outside the component, since it is just a helper function.